packages/synapse-react-client/src/lib/containers/markdown/UserMentionModal.tsx

```diff
@@ -26,19 +26,25 @@ export const UserMentionModal: React.FC<UserMentionModalProps> = ({
 
   return (
     <>
-      <Modal show={show} onHide={onClose} backdrop="static" animation={false}>
-        <Modal.Header closeButton>
-          <Modal.Title>Find User or Team</Modal.Title>
-        </Modal.Header>
-        <Modal.Body>
+      <DialogBase
+        open={show}
+        onCancel={onClose}
+        title="Find User or Team"
+        content={
           <UserSearchBoxV2
             placeholder="Search for a user or team name"
             onChange={onUserChange}
             typeFilter={TYPE_FILTER.ALL}
             focusOnSelect={true}
           />
-        </Modal.Body>
-      </Modal>
+        }
+        sx={{
+          '.MuiDialog-paperFullWidth': {
+            overflowY: 'visible',
+            '.MuiDialogContent-root': { overflowY: 'visible' },
+          },
+        }}
```

Comment:
Ensures the react-select dropdown menu is visible above the MUI dialog as described [here](https://stackoverflow.com/a/57946878). This does result in some styling changes to the TeamBadges, but the styling appears to more closely match the TeamBadge styling in other UserSearchBoxV2 instances, such as in the AccessSubmissionDashboard (i.e. no link styling applied to the TeamBadge) -- 
![main_ReviewerDashboard_SubmissionsReviewer](https://user-images.githubusercontent.com/26949006/236081200-5474b0fa-b937-42a4-8f5b-76078b239a64.png)


I also considered updating UserSearchBoxV2 directly as described [here](https://react-select.com/advanced#portaling) (i.e. adjust menuPortal z-index in styles and add `menuPortalTarget` argument), but the menu was moved outside the parent div of the component and this resulted in larger styling changes.